query/src/main/java/io/keen/client/java/Query.java

```diff
@@ -151,38 +181,61 @@ protected Query(Builder builder) {
      * individual queries may require additional arguments.
      */
     public static class Builder {
+        // required
         private QueryType queryType;
         private String eventCollection;
+        private Timeframe timeframe;
+
+        // sometimes optional
         private String targetProperty;
 
         // required by the Percentile query
         private Double percentile;    // 0-100 with two decimal places of precision for example, 99.99
 
         // optional
-        private Timeframe timeframe;
-        private List<Map<String, Object>> filters;
+        private Collection<Filter> filters;
         private String interval;
-        private String timezone;
-        private ArrayList<String> groupBy;
+        private List<String> groupBy;
         private Integer maxAge;
 
         public Builder(QueryType queryType) {
             this.queryType = queryType;
         }
 
         /**
-         * get filters
-         * @return a list of filters.
+         * Get filters
+         *
+         * @return a collection of filters.
+         */
+        public Collection<Filter> getFilters() { return filters; }
```

Comment:
Another case where we stepped half-way into the realm of major revision change, because Collection<> is more correct since List<> implies ordering, which I don't believe applies to filters. Also, Collection<Filter> seems more expressive and better hides internal details than WhateverCollection<Map<String, Object>> right? For a minor revision we can go back to the way it was.